layouts/layout-1d-base.js

```diff
@@ -189,12 +188,19 @@ export default class Layout extends EventTarget {
       this._emitRange();
       this._emitChildPositions();
     }
-    this._pendingReflow = null;
   }
 
   _scheduleReflow() {
     if (!this._pendingReflow) {
-      this._pendingReflow = Promise.resolve().then(() => this._reflow());
+      this._pendingReflow =
+          Promise.resolve().then(() => this.flushPendingReflow());
```

Comment:
This looks suspicious; why is it done? I think all these kinds of things need an extensive comment explaining why they're delayed instead of synchronous...

Reply:
Uhm, I guess we can now avoid doing batching here. That removes the need of `flushPendingReflow`.

The reason this is here is so that it doesn't matter which order we set layout properties like `direction, itemSize, viewportSize`. I'll look more into it, and at minimum add a comment there 👍

Reply:
So I played with this for a while, and if we make Layout reflow synchronously when a property changes, it becomes very important in which order properties are set. Also, reflow can be computationally expensive, so it doesn't make sense to trigger 3 reflows when the user sets 3 properties.

For coherency, I would keep microtask batching here and offer the `flushPendingReflow` method to force sync reflow.

In `<virtual-list>` I've removed the microtask batching, but I should probably put it back, for the same reasons.

Reply:
I am not very concerned with the performance when someone changes `layout`; that should be rare. I think the predictability of having only one batching strategy is more important.

I want to stress again that microtask batching is just not acceptable; it does not match anything on the platform. requestAnimationFrame based batching is the only realistic option.

An alternative you may want to consider is that, since all of this is delayed until requestAnimationFrame time anyway, you can reorder the changes as necessary to get best performance. But do so using normal code, not using microtask scheduling.

Reply:
Then what if we don't auto-reflow on property change, but just mark that we need to reflow?
The user will have to invoke `reflow()` themselves to apply the updates.

If yes, should we do the same for VirtualList? properties won't trigger render, but the user will have to invoke it themselves (in our case this will be done by `<virtual-list>`)

Reply:
Something like this:
```js
const layout = new Layout();
layout.updateItemSizes({...});
layout.reflow();

const list = new VirtualList();
list.totalItems = 10;
list.newChild = (i) => ...
list.layout = layout;
list.container = ...
list.render();

Reply:
I'm confused which user you are talking about. In general a strategy of marking what's needed so the next time the rAF callback comes around you do the reflow makes sense. But I don't want to require people to do `listElement.reflow()`.

Reply:
The `<virtual-list>` user wouldn't require to do `listElement.reflow()`, because `<virtual-list>` would handle internally rendering & reflow in its `[_render]()` function.

We have demoes that use `VirtualList` and `Layout` instances instead of `<virtual-list>` e.g. demo/basic/, demo/contacts/, demo/photos/

It feels weird for the `VirtualList/Layout` user that changing a property on `VirtualList` triggers dom updates, but changing properties on `Layout` does not.

The `<virtual-list>` user doesn't have access to the `Layout` instance, at least not yet.

Reply:
talked offline with @domenic, and agreed to have only Layout "passive" for now - in the sense that the user needs to call reflowIfNeeded() to trigger reflow.
Like this, the timing of the updates can be more predictable.

Reply:
Updated implementation and PR description 👌